### ISSUE TITLE
contribution must be created in Pmag GUI 3

### ISSUE THREAD

**Issue by moonshoes87** [PmagPy/PmagPy#155]
Previously Pmag GUI created a Contribution object either on __init__ or at the end of the convert to 3.0 step.  Now that the 3.0 conversion scripts are native, I need to make sure that the contribution is made once the 3.0 files are created -- after combine_magic has been run either just for measurements or for the 'other' file types.  Right now the contribution object isn't automatically updated/created at that time, so then Pmag GUI steps 2 and 3 don't work as expected.

**Comment by moonshoes87** [comment]
This is actually a thornier issue that I realized, and possibly the source of @ltauxe's problems with Thellier GUI.  In my code, I've tried to decrease the number of times we are reading in from csv files by passing the Contribution object around and updating it.  So right now, Pmag GUI writes to file often but reads from file rarely.  

I've refactored a bit to achieve the following: ba297513890e4a268b1f667c8782054b028370a3

- Pmag GUI now calls Thellier GUI and Demag GUI by creating a new frame instance, not by running main or start.  The Pmag GUI main frame is also now disabled while Thellier or Demag GUIs are running.

- Thellier GUI and Demag GUI main frames now both take an additional optional parameter, evt_quit.  This parameter is a custom event that fires when Thellier GUI or Demag GUI is closed.

- When Pmag GUI is pinged that Demag GUI or Thellier GUI have closed, it is re-enabled, shown, and also reads in any new data that might have been written to file by Demag or Thellier GUI.

Ideally, we wouldn't be reading in multiple times like this, but Demag and Thellier GUIs are both complicated and I think this is a good way (for now) to ensure that we don't miss things.  This update should also prevent some of the trouble @ltauxe had in #153, where I suspect Thellier GUI updated the files but not the Pmag GUI contribution object, thus resulting in some chaos.  

Next step: make sure that a contribution is created after the combine_magic step(s) in Pmag GUI.
